python-lib/dku_data_processing/filtering.py

```diff
@@ -0,0 +1,166 @@
+"""
+Filtering module used for applying filters on data in the backend.
+
+Handle a variety of different filtering techniques based on the type of the filter. The main
+function of the module is filter_dataframe(df, filters) that filter the DataFrame based on filters
+received from the front-end.
+
+"""
+
+import logging
+import pandas as pd
+
+from functools import reduce
+
+DKU_NO_VALUE = '___dku_no_value___'
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO, format='%(asctime)s Custom Chart Geospatial Density  | %(levelname)s - %(message)s')
+
+
+def numerical_filter(df, filter_):
+    """
+    Compute an boolean condition setting which rows of DataFrame `df` should remain based on
+    the numerical filter defined in the dict `filter_`.
+
+    :param df: Pandas DataFrame containing core data
+    :param filter_: Dictionary defining the filters
+    :return: conditions: An intermediary object containing stating if the row is kept or not
+    """
+    conditions = []
+    minimum_bound = filter_.get("minValue", None)
+    maximum_bound = filter_.get("maxValue", None)
+    if minimum_bound:
+        conditions += [df[filter_['column']] >= minimum_bound]
+    if maximum_bound:
+        conditions += [df[filter_['column']] <= maximum_bound]
+    return conditions
+
+
+def alphanum_filter(df, filter_):
+    """
+    Compute an boolean condition setting which rows of DataFrame `df` should remain based on
+    the alphanum filter defined in the dict `filter_`.
+
+    :param df: Pandas DataFrame containing core data
+    :param filter_: Dictionary defining the filters
+    :return: conditions: An intermediary object containing stating if the row is kept or not
+    """
+    conditions = []
+    excluded_values = []
+    for k, v in filter_['excludedValues'].items():
+        if k != DKU_NO_VALUE:
+            if v:
+                excluded_values += [k]
+        else:
+            if v:
+                conditions += [~df[filter_['column']].isnull()]
+    if len(excluded_values) > 0:
```

Comment:
I think it is a good practice to rather use the syntax:
```suggestion
    if excluded_values:
```
Indeed, if ever excluded_values is Null, it will not generate an error but just not enter into the condition. What do you think?

Reply:
Agreed and done in 4cf4276